lib/Agrammon/Web/Routes.pm6

```diff
@@ -213,24 +172,82 @@ sub routes(Agrammon::Web::Service $ws) is export {
 
         # test/implement
         post -> LoggedIn $user, 'new_tag' {
-                ...
             request-body -> (:$name!) {
-                my $data = $ws.new-tag($user, $name);
+                my $data = $ws.create-tag($user, $name);
                 content 'application/json', $data;
             }
         }
+    }
+}
 
-        ### input/output
-        
-        # test/implement (is this needed?)
-        post -> LoggedIn $user, 'get_input' {
-                ...
-            request-body -> (:$dataset!) {
-                my $data = $ws.get-input($user, $dataset);
+sub user-routes(Agrammon::Web::Service $ws) {
+    route {
+
+        # implement/test
+        post -> LoggedIn $user, 'change_password' {
+            request-body -> (:$old-password!, :$new-password!) {
+                my $data = $ws.change-password($old-password, $new-password);
+                content 'application/json', $data;
```

Comment:
Looking at this made me a bit curious: how does `$ws` know which user it is to change the password for? Should we be passing something to it? Same for various other operations.

Reply:
In `Agrammon::Web::SessionUser` I do set the `username` attribute to the login user.
What if I also set the `id` attribute as well? Then I can use the SessionUser everywhere I need the logged-in user.

Or is this a bad design?

Reply:
No, that seems sensible enough to me. I mean, carrying around huge amounts of user data in there can make the session expensive to save/load (though we can also lazily populate it), but I don't think one extra ID attribute is a problem at all.

Reply:
As it is currently implemented, upon authentication **all** user info is loaded into the SessionUser object.
As it is not all that much info and we don't have large numbers of users, I think we can leave it like that for the moment.

Reply:
With that setup, method calls on the SessionUser object will automagically work on the right database entry.